### PR TITLE
Bayesian Optimization

### PR DESCRIPTION
**Pull request by inventormc**
Add functionality for Bayesian optimization for searching over distributions.

### CODE REVIEW

File: tune_sklearn/tune_search.py [hunk 1]
```diff
@@ -888,6 +918,9 @@ def _fill_config_hyperparam(self, config):
                 configuration for `tune.run`.
 
         """
+        if self.search_optimization == "bayesian":
+            return
```
Comment (anthonyhsyu) [inline tune_sklearn/tune_search.py:922]: Not sure if this is right. I know you're passing this into `BayesOptSearch`'s `space` parameter which seems correct, but I'm not sure if leaving this out of the `config` dictionary would ruin anything. Just double check to make sure this is good!

File: tune_sklearn/tune_search.py [hunk 2]
```diff
@@ -1142,27 +1182,18 @@ def _tune_run(self, config, resources_per_trial):
...
-                scheduler=self.scheduler,
-                reuse_actors=True,
-                verbose=self.verbose,
-                stop={"training_iteration": self.early_stopping_max_epochs},
-                config=config,
-                checkpoint_at_end=True,
-                resources_per_trial=resources_per_trial,
-            )
+
+        analysis = tune.run(
```
Comment (anthonyhsyu) [inline tune_sklearn/tune_search.py:1188]: Nice refactor lol

File: tune_sklearn/tune_search.py [hunk 3]
```diff
@@ -840,25 +846,47 @@ class TuneRandomizedSearchCV(TuneBaseSearchCV):
...
+            raise ValueError("Search optimization must be random or bayesian")
+        if search_optimization == "bayesian" and random_state is not None:
+            warnings.warn(
+                "random state is ignored when using Bayesian optimization")
+
+        for dist in param_distributions.values():
+            if search_optimization == "random":
+                if not (isinstance(dist, list) or hasattr(dist, "rvs")):
+                    raise ValueError(
+                        "distribution must be list or scipy "
```
[the remark this answers is not in the document]
Comment (inventormc) [reply tune_sklearn/tune_search.py:888]: example has been added (bayesian_sgd.py)

File: tune_sklearn/tune_search.py (hunk 1 above)
Comment (inventormc) [reply tune_sklearn/tune_search.py:930]: This seems to be okay based on tune's example; the only parameter passed into config is iterations which doesn't look like a hyperparameter being tuned.

https://github.com/ray-project/ray/blob/master/python/ray/tune/examples/bayesopt_example.py

**Pull request merged**